packages/salesforcedx-vscode-core/src/diagnostics/diagnostics.ts

```diff
@@ -34,48 +34,47 @@ export const handlePushDiagnosticErrors = (
 ): vscode.DiagnosticCollection => {
   errorCollection.clear();
 
-  // In the case that we have deployed multiple source paths,
-  // the default error path for errors without an associated
-  // file path should be the workspace path
   const defaultErrorPath = sourcePathOrPaths.includes(',') ? workspacePath : sourcePathOrPaths;
 
   const diagnosticMap: Map<string, vscode.Diagnostic[]> = new Map();
-  if (Reflect.has(errors, 'files')) {
-    errors.files?.forEach(error => {
-      const fileUri = getFileUri(workspacePath, error.filePath, defaultErrorPath);
-      const range = getRange(error.lineNumber || '1', error.columnNumber || '1');
-
-      const diagnostic = {
-        message: error.error,
-        severity: vscode.DiagnosticSeverity.Error,
-        source: error.type,
-        range
-      } as vscode.Diagnostic;
-
-      if (!diagnosticMap.has(fileUri)) {
-        diagnosticMap.set(fileUri, []);
-      }
 
-      diagnosticMap.get(fileUri)!.push(diagnostic);
-    });
+  const processError = (filePath: string, lineNumber: string, columnNumber: string, error: string, type: string) => {
+    const fileUri = getFileUri(workspacePath, filePath, defaultErrorPath);
+    const range = getRange(lineNumber || '1', columnNumber || '1');
 
-    diagnosticMap.forEach((diagMap: vscode.Diagnostic[], file) => {
-      const fileUri = vscode.Uri.file(file);
-      errorCollection.set(fileUri, diagMap);
-    });
-  } else if (Reflect.has(errors, 'status')) {
-    const fileUri = vscode.Uri.file(defaultErrorPath);
-    const range = getRange('1', '1');
     const diagnostic = {
-      message: errors.message,
+      message: fixupError(error),
       severity: vscode.DiagnosticSeverity.Error,
-      source: errors.name,
+      source: type,
       range
     } as vscode.Diagnostic;
```

Comment:
you can change to `const diagnostic: vscode:Diagnostic` gets rid of the type assertion and lets the type checker check.

since fixupError always returns a string, it should satisfy the type